File: tags/svn+ssh:/svnuser@kelev.kaltura.com/usr/local/kalsource/backend/server/tags/eagle_2011_05_12/generator/sources/java/src/com/kaltura/client/KalturaParams.java
package com.kaltura.client;

import java.net.URLEncoder;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Set;

/**
 * Helper class that provides a collection of Kaltura parameters (key-value pairs).
 * 
 * @author jpotts
 *
 */
public class KalturaParams extends HashMap<String, String> {

	private static final long serialVersionUID = 6630046786691120850L;

	public String toQueryString() {
		String str = "";
		
		Set<String> s = this.keySet();
		Iterator<String> it = s.iterator( );
		while (it.hasNext()) {
			String name = (String)it.next();
			String value = (String)this.get(name);
			try {
				str += (name + "=" + URLEncoder.encode (value, "UTF-8") + "&");
			} catch (Exception e) {
				//TODO handle this
			}
		}
		
        if (str.endsWith("&"))
            str = str.substring(0, str.length() - 1);

        return str;        
	}
	
	public void add(String objectName, KalturaParams objectProperties) {
        for (String key : objectProperties.keySet()) {
            this.put(objectName + ":" + key, objectProperties.get(key));            
        }
    }

	public void add(KalturaParams objectProperties) {
		this.putAll(objectProperties);
    }

	public void setString(String key, String value) {
		if(this.get(key) != null)
			this.remove(key);
        this.put(key, value);
    }
	
	public void addObjectIfNotNull(String key, KalturaObjectBase object) {
		if (object == null)
			return;
		
		KalturaParams params = object.toParams();
		Set<String> s = params.keySet();
		Iterator<String> it = s.iterator( );
		while (it.hasNext()) {
			String name = (String)it.next();
			String value = (String)params.get(name);
			this.put(key + ":" + name, value);
		}
    }
	
	public void addObjectIfNotNull(String key, ArrayList array) {
		if (array == null)
			return;
		
		int index = 0;
		for(KalturaObjectBase object : (ArrayList<KalturaObjectBase>)array)
		{
			KalturaParams params = object.toParams();
			Set<String> s = params.keySet();
			Iterator<String> it = s.iterator( );
			while (it.hasNext()) {
				String name = (String)it.next();
				String value = (String)params.get(name);
				this.put(key + ":" + index + ":" + name, value);
			}
			index++;
		}
    }

	public void addStringIfNotNull(String key, String value) {
        if (value != null) this.put(key, value);
    }

	public void addIntIfNotNull(String key, int value) {
        if (value != Integer.MIN_VALUE) this.put(key, Integer.toString(value));
    }
	
	public void addFloatIfNotNull(String key, float value) {
        if (value != Float.MIN_VALUE) this.put(key, Float.toString(value));
    }
	
    public void addBoolIfNotNull(String key, boolean value) {
        this.put(key, value ? "1" : "0");
    }
}

File: tags/eagle_2011_08_11/generator/sources/csharp/KalturaClient/KalturaParams.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Web;

namespace Kaltura
{
    public class KalturaParams : SortedList<string, string>
    {
        public string ToQueryString()
        {
            string str = "";
            foreach (KeyValuePair<string, string> item in this)
                str += (item.Key + "=" + HttpUtility.UrlEncode(item.Value) + "&");

            if (str.EndsWith("&"))
                str = str.Substring(0, str.Length - 1);

            return str;
        }

        public void Add(string objectName, KalturaParams objectProperties)
        {
            foreach (KeyValuePair<string, string> item in objectProperties)
            {
                this.Add(objectName + ":" + item.Key, item.Value);
            }
        }

        public void Add(KalturaParams objectProperties)
        {
            foreach (KeyValuePair<string, string> item in objectProperties)
            {
                this.Add(item.Key, item.Value);
            }
        }

        public void AddStringIfNotNull(string key, string value)
        {
            if (value != null)
                this.Add(key, value);
        }

        public void AddIntIfNotNull(string key, int value)
        {
            if (value != int.MinValue)
                this.Add(key, value.ToString());
        }


        public void AddFloatIfNotNull(string key, float value)
        {
            if (value != Single.MinValue)
                this.Add(key, value.ToString());
        }

        public void AddEnumIfNotNull(string key, Enum value)
        {
            this.AddIntIfNotNull(key, value.GetHashCode());
        }

        public void AddStringEnumIfNotNull(string key, KalturaStringEnum value)
        {
            if (value != null)
                this.AddStringIfNotNull(key, value.ToString());
        }

        public void AddBoolIfNotNull(string key, bool? value)
        {
            if (value.HasValue)
                this.Add(key, (value.Value) ? "1" : "0");
        }
    }
}
